src/lib/y2partitioner/actions/controllers/fstabs.rb

```diff
@@ -267,8 +330,56 @@ def setup_blk_filesystem(filesystem)
           if filesystem.can_configure_snapper?
             filesystem.configure_snapper = filesystem.default_configure_snapper?
           end
+
           filesystem.setup_default_btrfs_subvolumes if filesystem.supports_btrfs_subvolumes?
         end
+
+        # Device indicated in the fstab entry
+        #
+        # When the device name in the fstab entry corresponds to a encryption device, the device
+        # could not be found by that fstab name. In general, the encryptions might be probed with
+        # a different name, so before searching for the device, the devicegraph is modified to
+        # set the encryption names from the crypttab file. That changes are made in a temporary
+        # devicegraph, so the original one is never altered.
+        #
+        # @see #devicegraph_with_fixed_encryptions
+        #
+        # @param entry [Y2Storage::SimpleEtcFstabEntry]
+        # @param devicegraph [Y2Storage::Devicegraph]
+        #
+        # @return [Y2Storage::BlkDevice, Y2Storage::Filesystems::Nfs, nil] nil if the device is
+        #   not found in the devicegraph.
+        def entry_device(entry, devicegraph)
+          device = entry.device(devicegraph_with_fixed_encryptions(devicegraph))
+          device ? devicegraph.find_device(device.sid) : nil
+        end
+
+        # Duplicates the given devicegraph and modifies it by setting the encryption names
+        # from the crypttab file
+        #
+        # @see #crypttab
+        #
+        # @param devicegraph [Y2Storage::Devicegraph]
+        # @return [Y2Storage::Devicegraph]
+        def devicegraph_with_fixed_encryptions(devicegraph)
+          fixed_devicegraph = devicegraph.dup
+
+          return fixed_devicegraph unless crypttab
+
+          Y2Storage::Encryption.use_crypttab_names(fixed_devicegraph, crypttab)
+          fixed_devicegraph
+        end
+
+        # Selects the crypttab contained in the same filesystem than the selected fstab
+        #
+        # @return [Y2Storage::Crypttab, nil] nil if the filesystem does not contain
+        #   a crypttab file.
+        def crypttab
+          return @crypttab if @crypttab_found
```

Comment:
Hmmm. Does this mean the crypttab is not refreshed even if `#selected_fstab` changes? Looks like a bug to me.

Reply:
What about creating a unit test case for that corner case? Just in case some other developer feel tempted to cache that in the future. Only if it's easy enough, of course.